src/base/bittorrent/addtorrentparams.h

```diff
@@ -52,7 +55,7 @@ namespace BitTorrent
         TriStateBool addPaused;
         QVector<DownloadPriority> filePriorities; // used if TorrentInfo is set
         bool skipChecking = false;
-        TriStateBool createSubfolder;
+        boost::optional<BitTorrent::TorrentContentLayout> contentLayout;
```

Comment:
Is `TriStateBool` unable to handle our use case here?

Also IIRC `std::optional` in C++17 has some minor differences to `boost::optional`, are we able to replace boost easily when we have C++17?

Reply:
>Is TriStateBool unable to handle our use case here?

No, of course. Now we need 3 values plus extra one to indicate "value is not exactly set".

>are we able to replace boost easily when we have C++17?

What could be the problem?
I was going to use it initially, but we're still on C++14... So I'll change it as soon as we move to C++17.